src/model/application.rs

```diff
@@ -89,6 +91,38 @@ pub struct CurrentApplicationInfo {
     #[serde(default)] pub rpc_origins: Vec<String>,
     pub bot_public: bool,
     pub bot_require_code_grant: bool,
+    pub team: Option<Team>,
     #[serde(skip)]
     pub(crate) _nonexhaustive: (),
 }
+
+/// Information about the Team group of the application.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Team {
+    /// The icon of the TEam.
+    pub icon: Option<String>,
+    /// The snowflake ID of the team.
+    #[serde(deserialize_with = "deserialize_u64")]
+    pub id: u64,
+    /// The members of the team
+    pub members: Vec<TeamMember>,
+    /// The user id of the team owner.
+    pub owner_user_id: UserId,
+}
+
+/// Infromation about a Member on a Team.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct TeamMember {
+    /// The member's membership state.
+    ///
+    /// 1: Invited
+    /// 2: Accepted
+    pub membership_state: u8,
```

Comment:
> `TeamMember::membership_state` could be an enum, but i couldn't figure out how to make it deserializable.

We have a macro built for this purpose, `enum_number!`. Here's an example of its usage:
https://github.com/serenity-rs/serenity/blob/current/src/model/gateway.rs#L357-L396

Reply:
Resolved on ec0286b.